config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -155,16 +164,21 @@ public void deleteConfigs(
         return;
       }
 
-      Set<ConfigResourceContext> configResourceContexts =
+      List<ConfigResourceContext> configResourceContexts =
           request.getConfigsList().stream()
               .map(this::getConfigResourceContext)
-              .collect(Collectors.toUnmodifiableSet());
-      Map<ConfigResourceContext, ContextSpecificConfig> configs =
-          configStore.getAllContextConfigs(configResourceContexts);
+              .collect(Collectors.toUnmodifiableList());
+      Map<ConfigResourceContext, Optional<ContextSpecificConfig>> configs =
+          configStore.getContextConfigs(configResourceContexts);
```

Comment:
Do you need to return a Map<context, optional<config>> ? Why not just remove the contexts that aren't present from the map entirely. Then you can delete only the keyset to save a bit of work.

Reply:
We returned this because there is dependency of empty config in multiple places across config service. so Optional<ContextSpecificConfig> will sometimes have non empty value but config inside is empty. I didn't filter it out because it might lead to missing some configs which are null and present in DB

Reply:
So I don't think we should mix optional and empty config (if I'm understanding you correctly here). If we're switching to optionals, we should be replacing the empties. We don't want two ways to represent "no config". It might make sense to back out the change from empty -> optional and do that in a separate PR.

Reply:
let me track this via separate JIRA. I will create a separate PR for this. That might be big change in terms of how we handle configs right now in terms of creation and updates